### ISSUE TITLE
Add `executionConfig` to `context`, and a `loadExecutionConfig` lifecycle method that leverages the `instance.config`

### ISSUE THREAD

**Issue by ndowmon** [JupiterOne/sdk#583]
```ts
function executeIntegration(context, invocationConfig) {
  if (invocationConfig.loadExecutionConfig) {
    context.executionConfig = invocationConfig.loadExecutionConfig(context.instance.config); // THIS SHOULD NOT BE ASYNC
  else { 
    context.executionConfig = context.instance.config;
  }
  ...
}
```

**Comment by austinkelleher** [comment]
Can you elaborate on the use case? We want to avoid bleeding too many "JupiterOne integration" implementation details into the SDK.

**Comment by ndowmon** [comment]
The main use case is producing credentials chains that can be passed from one step to the next, rather than forcing each step to make its own call to fetch (or refresh) its token.

**Comment by aiwilliams** [comment]
I think the main use case is to calculate configuration for execution, to avoid mutating the `instance.config`, and to avoid doing so in functions like `validateInvocation` and `getStepStartStates`. See https://github.com/JupiterOne/graph-qualys/blob/d026da38f4963b751283c09dc7a2b0f2b4c91121/src/validateInvocation.ts#L43 for an example (antipattern).

**Comment by aiwilliams** [comment]
This issue is being addressed by https://github.com/JupiterOne/sdk/pull/587.
I've got another use case in https://github.com/JupiterOne/graph-jira/pull/125.

**Comment by ceelias** [comment]
Closed by #587
